test/widgets/emoji_reaction_test.dart

```diff
@@ -556,6 +556,32 @@ void main() {
         debugNetworkImageHttpClientProvider = null;
       });
 
+      testWidgets('text emoji list items must have correct height', (tester) async {
+        // Regression test for #1587
+        tester.platformDispatcher.textScaleFactorTestValue = 2.0;
+        addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
+
+        final message = eg.streamMessage();
+        // Set up the emoji picker, which leaves it open.
+        await setupEmojiPicker(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+
+        // Now, change the setting to text-only emoji.
+        await store.handleEvent(UserSettingsUpdateEvent(id: 1,
+          property: UserSettingName.emojiset,
+          value: Emojiset.text));
+        // The picker is already open and will rebuild with the new setting.
+        await tester.pumpAndSettle();
+
+        // Find a list entry and check its height.
+        final listEntryFinder = find.byType(EmojiPickerListEntry).first;
+        await tester.ensureVisible(listEntryFinder);
+
+        final renderBox = tester.renderObject<RenderBox>(listEntryFinder);
+        check(renderBox.size.height).isLessThan(45.0);
```

Comment:
Why check that the height is _less than_ a number? The point of this test is to make sure the height is large enough. In other words (as you've written in a comment at the top), it's a regression test for #1587.